src/Files.App/DataModels/NavigationControlItems/LocationItem.cs

```diff
@@ -52,11 +54,39 @@ public bool IsExpanded
 		}
 
 		public bool IsInvalid { get; set; } = false;
-
+		public bool IsRecycleBin => string.Equals(Path, "shell:RecycleBinFolder", StringComparison.OrdinalIgnoreCase);
 		public SectionType Section { get; set; }
 
 		public ContextMenuOptions MenuOptions { get; set; }
 
 		public int CompareTo(INavigationControlItem other) => Text.CompareTo(other.Text);
 	}
+
+	public class RecycleBinLocationItem : LocationItem
```

Comment:
I think we can avoid this if we tweak line 37.

Reply:
Yes, but that would imply putting the refresh function inside LocationItem, which would be somewhat out of place because it's only for a specific instantiation of a LocationItem